pkg/driver/helmdriver.go

```diff
@@ -22,19 +22,18 @@ import (
 
 // helmDriver implements PackageDriver to install packages from Helm charts.
 type helmDriver struct {
-	cfg      *action.Configuration
-	log      logr.Logger
-	settings *cli.EnvSettings
+	cfg        *action.Configuration
+	secretAuth auth.Authenticator
+	log        logr.Logger
+	settings   *cli.EnvSettings
 }
 
 var _ PackageDriver = (*helmDriver)(nil)
 
-func NewHelm(log logr.Logger) (*helmDriver, error) {
+func NewHelm(log logr.Logger, secretAuth auth.Authenticator) (*helmDriver, error) {
 	settings := cli.New()
 
-	// TODO Catch error here if not provided docker config or continue without an authfile if possible
-	secretAuth := auth.NewHelmSecret()
-	authfile, _ := secretAuth.AuthFilename()
+	authfile := secretAuth.AuthFilename()
 	if authfile != "" {
```

Comment:
Github won't let me suggest this via their web editor, cuz they're annoying, but...

Lines 37-39 are doing nothing. ClientOptCredentialsFile has no side-effects when called. It returns a ClientOpt (which itself is a function), and that ClientOpt should be passed to the registry.NewClient call on line 40. It should look more like this:

```
client, err := registry.NewClient(registry.ClientOptCredentialsFile(authfile))
```

Which all begs the question of if you really need to set the credentials file at all? As I assume this code was working before now..? I guess it got missed in previous reviews? Or am I just missing something?

I'm kind of suprised one of our many linters doesn't catch this.

Reply:
fixed